src/doc/tock_binary_format.md

```diff
@@ -586,6 +587,37 @@ Object has both, the kernel decides which to use. Tock is transitioning to
 having the Program Header as the standard one to use, but older kernels (2.0 and
 earlier) do not recognize it and use the Main Header.
 
+#### `10` ShortID
+
+```rust
+struct TbfHeaderV2ShortId {
+    base: TbfHeaderTlv,
+    short_id: u32,
```

Comment:
```suggestion
    short_id: core::num::NonZeroU32
```

Reply:
I think it makes sense to keep it `u32`, since that is what is actually stored in the header. This describes the meaning of the bytes in the header more than the semantic meaning of the bytes.

Reply:
I don't really feel strongly about this, so feel free to just ignore me and hit resolve, but given the decision above to delete the statement about 'what happens if `0` (an invalid Short ID) is provided`, that would make putting `0` in the header undefined behavior -- much like how using unsafe to force zero in the memory underlying a `NonZeroU32` is undefined behavior. Arguably, `NonZeroU32` better reflects what's stored in the header, well defined values for all values not `0` [ One of those comments where after writing, one realizes the PL folks are seeping into their brain..]

Reply:
I see your point, but this isn't defining a programming language it is specifying the bitwidths of fields in a header.